### ISSUE TITLE
Introduce `final_states_cont` and `root_states_cont`

### ISSUE THREAD

**Issue by Marius1311** [theislab/cellrank#111]
Additionally to the categorical `root_states` and `final_states`, we need continious `root_states_cont` and `final_states_cont` - basically, the probability of being a root/final states, irrespective of _which one_. 

For CFLARE, this already exists, I guess it's currently called `final_cells_probs` and `root_cells_probs`, so here, it's only a matter of renaming. For GPCCA, we can get this simply by adding up our lineage probabilities corresponding to root/final states (not taking into account the `rest` in case we did not redistribute, otherwise it won't work). 

Once this is done, we can nicely interface with scvelo's latent time and directed paga. We're talking to Volker tomorrow, would be nice to have this ready by then so he can start with the interface on scvelo's side.

**Comment by michalk8** [comment]
I'm not sure if i follow you correctly for `probs` in GPCCA - you want so sum up the lineages
for the main states except for rest, or do you want to do this only for some restricted cells?
If you want them sum up, I don't think it will work, see below (Alpha, Beta, Epsilon):
![sum](https://user-images.githubusercontent.com/46717574/81741361-be7f9f80-949e-11ea-82e3-f8399ec5da1d.png)

Max seems to work better:
![max](https://user-images.githubusercontent.com/46717574/81741352-ba538200-949e-11ea-9b41-32794e11dc03.png)

And for Alpha and Epsilon only:
![ae](https://user-images.githubusercontent.com/46717574/81741466-e7079980-949e-11ea-914f-a8d962564191.png)

**Comment by michalk8** [comment]
Note that I haven't normalized so that it sums to 1, but with `sum` over the lineages, the picture is almost the same.
Maybe more robust way instead of `max` could be `var` - if a cell is in a metastable state, the variance of probs should be big (not uniform). Here's how it would look (now normalized to sum to 1):

![var](https://user-images.githubusercontent.com/46717574/81742187-fa673480-949f-11ea-9090-10cd8bbbdffa.png)

If I did misunderstand, please correct me.

**Comment by Marius1311** [comment]
Thanks Mike, you are right, simply summing them up is not a good idea, as you show above. 

What do you mean by `max`? Can you post your code?

**Comment by Marius1311** [comment]
@VolkerBergen @michalk8 should we go for `root_states_probs` and `final_states_probs`, or simple `root_probs` and `final_probs`? Simpler and shorter, no?

**Comment by michalk8** [comment]
Just for reference, I've plotted how `P(A u B u C) = P(A) + P(B) + P(C) - P(A ^ B) - ...` would look like (in code using the max as discussed). Code for reference:
```python
from itertools import combinations

x = adata.obsm['to_final_states']
x = x['Alpha', 'Beta', 'Epsilon']

current = x.sum(1)
for c in combinations(range(x.shape[1]), 2):
    val = np.prod(x[:, c], axis=1)
    current -= val
current += np.prod(x, axis=1)

adata.obs['P(A u B u C)'] = current.X.squeeze()
adata.obs['diff'] = adata.obs['P(A u B u C)'] - adata.obs['final_states_probs']
scv.pl.scatter(adata, color=['final_states_probs', 'P(A u B u C)', 'diff'])
```
![index](https://user-images.githubusercontent.com/46717574/81994582-74d2b880-9648-11ea-902b-1c7dbda721d8.png)

I'd keep it `root_state_probs` - it's a bit more descriptive and not much longer (+ we have `to_final_states` and `final_states`).

**Comment by Marius1311** [comment]
Just for clarification: you compare `P(XOR(A, B, C))` with using the max? Or how it `final_states_probs` computed above?
